conda/common/cuda.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+
+def cuda_detect():
+    '''Attempt to detect the version of CUDA present in the operating system.
+    
+    On Windows and Linux, the CUDA library is installed by the NVIDIA 
+    driver package, and is typically found in the standard library path,
+    rather than with the CUDA SDK (which is optional for running CUDA apps).
+    
+    On macOS, the CUDA library is only installed with the CUDA SDK, and
+    might not be in the library path.
+    
+    Returns: version string (Ex: '9.2') or None if CUDA not found.
+    '''
+    # platform specific libcuda location
+    import platform
+    system = platform.system()
+    if system == 'Darwin':
+        lib_filenames = [
+            'libcuda.dylib', # check library path first
+            '/usr/local/cuda/lib/libcuda.dylib'
+        ]
+    elif system == 'Linux':
+        lib_filenames = [
+            'libcuda.so', # check library path first
+            '/usr/lib64/nvidia/libcuda.so', # Redhat/CentOS/Fedora
+            '/usr/lib/x86_64-linux-gnu/libcuda.so', # Ubuntu
+        ]
```

Comment:
This is the driver libcuda, and not the cuda runtime. This will return the latest version of CUDA supported as opposed to the version in use.